MotionMark/resources/statistics.js

```diff
@@ -223,6 +223,58 @@ Regression = Utilities.createClass(
         return this.s1 + this.t1 * complexity;
     },
 
+    _intersection: function(segment1, segment2)
+    {
+        return (segment1.s - segment2.s) / (segment2.t - segment1.t);
+    },
+
+    _error: function() {
+        return this.e1 + this.e2;
+    },
+
+    _areEssentiallyEqual: function(n1, n2) {
+        const epsilon = 0.0001;
+        return Math.abs(n1 - n2) < epsilon;
```

Comment:
I just wanted to avoid having two essentially parallel lines but treated as intersected lines. This will lead to having a very far intersection point by this calculation if the difference between segment2.t and segment1.t is too small.
```
 _intersection: function(segment1, segment2)
{
    return (segment1.s - segment2.s) / (segment2.t - segment1.t);
}
```